src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md

```diff
@@ -114,29 +125,253 @@ Template:
+| Variable flip angle                        | VFA     | Grouping | The VFA method involves at least two spoiled gradient echo (SPGR) of steady-state free precession (SSFP) images acquired at different flip angles. Depending on the provided metadata fields and the sequence type, data may be eligible for DESPOT1, DESPOT2 and their variants ([Deoni et al. 2005](https://onlinelibrary.wiley.com/doi/full/10.1002/mrm.20314)). Please visit the [qMRI appendix](#prioritylevels) for details. _Associated output suffixes_: T1map, T2map, R1map, R2map |
+| Inversion recovery (for T1 mapping)        | IRT1    | Grouping | The IRT1 method involves multiple inversion recovery spin-echo images acquired at different inversion times ([Barral et al. 2010](https://onlinelibrary.wiley.com/doi/full/10.1002/mrm.22497)). _Associated output suffixes_: T1map, R1map                                                                                                                                                                                                                                                                                                                    |
+| Magnetization prepared two gradient echoes | MP2RAGE | Grouping | The MP2RAGE method is a special protocol that collects several images at different flip angles and inversion times to create a parametric T1map by combining the magnitude and phase images ([Marques et al. 2010](https://www.sciencedirect.com/science/article/pii/S1053811909010738?casa_token=u_CYBx4hi7IAAAAA:3w0cMTyU5jA1BdFs0s5oVcQeqF2tZho0iJ9d4N1kExfaX27v9-JnWacF6mbEp_lMKZ64CvoTl8k)). _Associated output suffixes_: T1map, R1map, UNIT1                                                                                                                                                                                                                 |
+| Multi-echo spin echo                       | MESE    | Grouping | The MESE method involves multiple spin echo images acquired at different echo times and is primarily used for T2 mapping. _Associated output suffixes_: T2map, R2map, MWFmap                                                                                                                                                                                                                                                                                                                         |
+| Multi-echo gradient echo                   | MEGRE   | Grouping | Anatomical gradient echo images acquired at different echo times. _Associated output suffixes_: T2starmap, R2starmap                                                                                                                                                                                                                                                                                                                                             |
```

Comment:
What is the distinction between `MEGRE` and `GRE` (e.g., as proposed in [BEP-004](https://docs.google.com/document/d/1kyw9mGgacNqeMbp4xZet3RnDhcMmf4_BmRgKaOkO2Sc/edit?pli=1) with an echo entity?

Reply:
This came up before. I think the reason we introduce `MEGRE` is that a single-echo GRE is usually meant to be either T1-, T2*, or PD-weighted, without any further qMRI-ish processing. Thus, a data set curator can communicate the weighting and intended use-case using the suffixes `_T1w`, `_T2starw`, and `_PDw`.

For the multi-echo GREs, you usually move from a PDw to a T2starw with longer echo times in a single scan. Thus, most of those images are neither T2*w, nor PDw. Moreover, you usually want to combine them in further computational processing pipelines (e.g., make a `T2starmap`). They are a typical case of a "grouped scan collection".

However, you make a valid point. The multi-echo part can already be communicated by having `echo-` key/value pairs. I'm personally not married to the `_MEGRE`-solution. @tsalo, what do you think about this now?

Reply:
The same applies to `MESE`.

Reply:
As @Gilles86 mentions, `MEGRE `and `MESE` are Grouing rather then Conventional suffixes, so they denote certain processing. As such they are helpful for tool developers rather than end-users. I also agree that there is a redundancy here.

Reply:
@tiborauer
>  perhaps all grouping suffixes could be omitted and substituted by a combination of entities.

I don't think that there is a way to omit grouping suffixes altogether and solve the problem with a combination of entities. Neither for more  involved descriptions such as `MPM`, nor for the relatively simplistic `GRE` and `MEGRE` case. 

We may assume that `sub-01_echo-1_GRE.nii.gz` and `sub-01_echo-1_MESE.nii` would be interchangeable, until there is a `sub-01_GRE.nii.gz` is present in the same dataset. Whereas `MESE` ensures that the file attains an `echo` entity and does not orphan any files. 

The usefulness of `GRE` as a suffix, I think, is a discussion of its own. Suffixes were meant to identify images with similar contrasts across datasets. As @Gilles86 pointed out, `sub-N_GRE.nii.gz` from `ds-001` can be different than  `sub-N_GRE.nii.gz` from ds-002` in terms of weighting, whereas `T1w`, `PDw` etc. can resolve such violation of interchangeable contrast.

Reply:
@agahkarakuzu
I agree with you. I am a fan of the groupping suffixes, but I want(ed) to point out that they serve a different purpose than Conventional suffixes, which are "were meant to identify images with similar contrasts across datasets". We should, perhaps, call Conventional suffixes Contrast suffixes.

From the POV of representing contrast, `sub-01_echo-1_GRE.nii.gz` and `sub-01_echo-1_MEGRE.nii` could potentially represent the very same data. So there is a redundancy for this use case, and the **data producing user** may be confused which to use.
On the other hand, from the POV of representing processing (requirement), `sub-01_echo-1_GRE.nii.gz` and `sub-01_echo-1_MEGRE.nii` are not equivalent for several reasons:
- As you pointed out, contrast MUST NOT be automatically assumed fro MEGRE
- MERGE MAY/SHOULD be processed (and in combination with the other images with MEGRE suffix).
So for a **data consumer user** , these suffixes are highly informative and even instructuctive.

Reply:
> For the multi-echo GREs, you usually move from a PD-weighted to a T2*-weighted image with longer echo times in a single scan. Thus, most of those images are neither T2*-, nor PD-weighted. Moreover, you usually want to combine them in further computational processing pipelines (e.g., make a T2starmap). They are a typical use-case of a "grouped scan collection".

@Gilles86 Thank you. I can see the use of "grouped scan collections" for specific protocols, although I don't think naming them according to the varying parameter is good. For example, VFA seems to have more requirements than just a varying flip angle, but you could have another scan type with a varying flip angle and it's unclear what you would name it given how generic the name "VFA" is. 

However, if the only distinction for a given suffix _is_ the varying parameter (as with MESE, MEGRE, MTS, and MPM), then it seems like they should just use entities and some generic suffix (e.g., SE, GRE) to delineate themselves.

> We may assume that `sub-01_echo-1_GRE.nii.gz` and `sub-01_echo-1_MESE.nii` would be interchangeable, until there is a sub-01_GRE.nii.gz is present in the same dataset. Whereas MESE ensures that the file attains an echo entity and does not orphan any files.

@agahkarakuzu Can you clarify what you mean by orphaning files? 

I'm trying to extend this to what I'm already familiar with, such as single-echo and multi-echo EPI data. I understand that there is a crucial difference between EPI and MEGRE data here, in that with EPI there is the assumption of interchangeability. You can analyze a single echo from EPI data and, while it might not be the _optimal_ echo time, it should still work, while the same isn't true of MEGRE. At least that's my understanding based on the BEP.

Still, that doesn't explain to me why one would need grouping suffixes instead of generic suffixes in conjunction with entities. If we want to estimate T2* from EPI data with `tedana`, then we need multi-echo EPI data. `tedana` can't run on single-echo data, so from a tool developer perspective we do need some way to identify appropriate files, but all we need to do is check for the `echo` entity with the `bold` suffix. Whether multi-echo-specific processing is necessary or optional for a given output doesn't necessarily need to be a concern for the specification, and could be documented in the qMRI documentation also included in this PR (e.g., the `Table of method-specific priority levels for qMRI metadata` in Appendix 9).

Let's say you acquire GRE data with both multiple echoes and multiple flip angles. You should be able to run the same pipeline on it as you would on MEGRE data, as long as you select the most appropriate `fa` files, but it's unclear to me whether you would still use MEGRE or need a new suffix. 

> I agree with you. I am a fan of the groupping suffixes, but I want(ed) to point out that they serve a different purpose than Conventional suffixes, which are "meant to identify images with similar contrasts across datasets". We should, perhaps, call Conventional suffixes Contrast suffixes.

@tiborauer Absolutely, I love that idea! That does make the distinction much clearer- at least for reflecting the assumption of interchangeability.

I think "Grouping" would also be clearer as "Non-Contrast" or something. For me, the term "grouping suffix" is confusing. Any suffix can refer to a group of files acquired at the same time, as it's the entities that determine whether data from the same scan is spread across multiple files or not. Some entities represent information from the same acquisition (e.g., `echo`, `part`, `fa`, and hopefully `ch` at some point), while others separate acquisitions (e.g., `sub`, `ses`, `run`). The entities that do represent when data from the same scan are split across multiple files can be applied to many suffixes that I'm sure you wouldn't count as "grouping," like `_bold` (even though you _could_ use multi-echo EPI data to estimate T2* in seconds, matching with qMRI).

The distinction is whether there's an inherent contrast or if you need multiple files to produce one, right? So "Non-Contrast" or "Contrast-Free" or something similar makes more sense to me.

Reply:
+1 for "Contrast" and "Non-Contrast"

Reply:
@tsalo 

> For example, VFA seems to have more requirements than just a varying flip angle, but you could have another scan type with a varying flip angle and it's unclear what you would name it given how generic the name "VFA" is.

`VFA` is a commonly used term (at least in ISMRM community), it is mostly referred to as `VFA framework` (a [quick reference](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5082493/) I hit when I search the term). Depending on the sequence type, or some other parameters that are not necessarily varying, the outputs of the `VFA` application changes (e.g. DESPOT1, DESPOT2, MCDESPOT, DAM ,...).  

> However, if the only distinction for a given suffix is the varying parameter

There are different conventions to the naming of the qMRI methods. Even when they have relatively more generic names, such as `VFA` , they denote the qMRI application by means of the sequence (if the sequence is designed for quantitative mapping [`MP2RAGE`, `SA2RAGE`], altered parameters [`double angle method`, `VFA`], type of outputs [`MPM`], or special names  [`Bloch-Siegert B1 mapping`, `DESPOT1-HIFI`, `AFI` etc.].

> Still, that doesn't explain to me why one would need grouping suffixes instead of generic suffixes in conjunction with entities. 

Let's come up with a generic suffix name that can apply to a variety of qMRI methods: `FLASH`, or `GRE`, does not matter (just misses the point that the sequences are spoiled). If we don't bother that `FLASH` is vendor specific for the sake of the example, the following cases should make clear why generic names can get easily confusing. 

i)  Double angle B1 mapping 
```
sub-01_fa-1_FLASH.nii.gz
sub-01_fa-2_FLASH.nii.gz
```
OK, these inputs would go into the `fmap` folder, but in future a new method would create a similar duality in the `anat` folder.

ii) VFA T1 mapping  
```
sub-01_fa-1_FLASH.nii.gz
sub-01_fa-2_FLASH.nii.gz
```

iii) MPM
```
sub-01_echo-1_fa-1_mt-on_FLASH.nii.gz
sub-01_echo-1_fa-1_mt-off_FLASH.nii.gz
sub-01_echo-1_fa-2_mt-off_FLASH.nii.gz
```
Note that not necessarily multi- echo. 

iv) MTsat
```
sub-01_fa-1_mt-on_FLASH.nii.gz
sub-01_fa-1_mt-off_FLASH.nii.gz
sub-01_fa-2_mt-off_FLASH.nii.gz
```
This can be multi-echo as well. 

v) MTR
```
sub-01_fa-1_mt-on_FLASH.nii.gz
sub-01_fa-1_mt-off_FLASH.nii.gz
```

These are just a few examples off the top of my head, that are going to overlap with each other. That's why we need grouping suffixes for sensible delineation of these input files. Many qMRI methods revolve around a similar frameworks. Sometimes one method differs from another at the pulse sequence level (e.g. `MTsat` vs `MPM`) that may not be even easily infer looking at the json metadata, let alone the fact that it hampers human readability. 

> Let's say you acquire GRE data with both multiple echoes and multiple flip angles. You should be able to run the same pipeline on it as you would on MEGRE data, as long as you select the most appropriate fa files, but it's unclear to me whether you would still use MEGRE or need a new suffix.

Examples `ii` and `iii` above should clarify this to some extent. And the overlap would increase over time as new methods are developed. Files of two methods can be namesake when named with a generic suffix, and those methods may even appear like identical twins (`MPM` and `MTsat`), but they need different IDs.

But if two methods share the exact same framework and differ from each other depending on which parameters you alter (this is not the case for MPM and MTsat), then `qMRI application that can be derived from an existing suffix` becomes practical to describe inputs without the need of introducing a new suffix (`VFA` and `DESPOT` family). 

> Can you clarify what you mean by orphaning files?

All the files identified by a groping suffix are going to be taken as an input for processing, no file sharing a common suffix is going to be left out alone (sounded dramatic). But this is not going to be the case if you use `GRE` to identify both a high res `T1w` that is to be used for surface extraction, and multi-echo `GRE` in the same directory. That `T1w` image won't join the qMRI party, but still be there. Mostly confusing for a human reader, and not ideal for picking up files. 

@tsalo I did my best to address your points (around 1am in Turkey), happy to clarify more if what I said above does not make sense. 

I can see why `contrast` and `non-contrast` sounds like a good idea. On the other hand, a negating expression would not be really informative. I don't find declaring "what it is not for" really useful for explaining which function do they serve. This would also create a semantic obstacle for people who would like to use one of the images (e.g. higher FA, mt-off of MPM) outside qMRI pipelines.

Reply:
@agahkarakuzu Thank you very much for explaining things in so much detail, and I'm sorry for the delay in responding in the PR.

I think there is a strong need for new suffixes for quantitative MRI protocols, and I agree that the concept of grouping multiple data acquisitions under the same suffix is necessary for qMRI as well.

However, I’d like to propose a compromise that I believe supports qMRI while fitting in with the current specification better, with the following changes:
1. A new concept for grouping multiple data acquisitions under the same suffix. This could be a concept rather than a suffix category, to allow for single-acquisition and multi-acquisition cases to share the same suffix. So we don’t need to assume that MPM scans, for example, came from separate acquisitions.
2. A new rule that files which only vary in multi-contrast entities (e.g., `echo`, `part`) must be associated. This will require clarifying language in the specification regarding these kinds of entities, and adding the caveat that, in cases where unassociated files differ only in these entities, the files should have at least one other (non-multi-contrast) entity to differentiate them (like `acq` or `run`).
    - This would resolve something that @tiborauer and @agahkarakuzu noted [in Mattermost](https://mattermost.brainhack.org/brainhack/pl/pc6ed7f9bfddbbohc8mminfubo), which is that, the way `run` is currently defined, if scans vary by a parameter (e.g., `echo`), then `run` should not be incremented. So, for example, we wouldn't know if `sub-X_task-Y_echo-1_bold.nii.gz` and `sub-X_task-Y_echo-2_bold.nii.gz` came from the same acquisition of not.
3. No new suffixes should be added that can be sufficiently dissociated based on the presence of entities. Thus, new suffixes are needed for frameworks with specific parameter values (e.g., `VFA`) but not for cases where a more general suffix combined with an entity would suffice (e.g., `MEGRE`). This prevents unnecessary redundancy, and tools for interacting with BIDS datasets (e.g., `pybids`) should be able to filter by the presence of entities and group associated files together easily.
    - @agahkarakuzu, this allows you to address the issue you raised above about distinguishing between protocols that only vary at the pulse sequence level by adding necessary suffixes, but while still making full use of the mechanisms already available in BIDS.

Reply:
I agree with 1.
For 2., as far as I understand you suggest using the "non-multicontarst" entity as a grouping element? Would not be more consistent to use groping suffix? Otherwise, one might ask why do not we use the "non-multicontarst" entity for 1., too.
For 3., I am not sure what is the difference between VFA and MEGRE from your POV. Moreover, the issue identified in 2. also applies for MEGRE and MESE.

Reply:
My impression from the derivatives discussion is that -- in general -- we'd like to avoid introducing many new suffixes, or heavily relying on them, as these can generally sprawl outwards somewhat indefinitely (though @effigies should correct me if that doesn't match his understanding). Because of that, I agree with @tsalo that it would be more consistent with the current spec to avoid new suffixes to the fullest extent possible.

Of course, there seem to be cases in qMRI where a new grouping suffix is absolutely necessary, and can't be disambiguated any other way. These seems to be the case of (1). But for ME-EPI, for example, I'd be against a new suffix. In that case, exactly as @tsalo suggested, we could keep the `_bold` suffix with the `echo` key-value pair, and that would sufficiently disambiguate the usage.

Reply:
I think the conclusion ended up being "we should try to consider when we can use one suffix + a parameterized entity instead of N suffixes, but suffix proliferation is not entirely containable". As a example, the [functional derivative maps](https://bids-specification.readthedocs.io/en/bep012/05-derivatives/05-functional-derivatives.html#functional-derivatives-maps) ended up being a cluster of suffixes.

Reply:
How do you distinguish the case of two EPI runs with different echo times from the case of ME-EPI with two echos?

Reply:
`acq` is often used for this, though maybe it's a bit unusual for the entire difference in acquisition parameters to be something that has its own entity.

```
sub-01_acq-echo1_bold.nii.gz
sub-01_acq-echo2_bold.nii.gz
```

I don't think that indices have to form an obvious sequence, holding all other entities equal, so I would assume this would work:

```
sub-01_run-1_echo-1_bold.nii.gz
sub-01_run-2_echo-2_bold.nii.gz
```

I don't know if the validator would warn, but PyBIDS would certainly handle this smoothly.

Reply:
The question is how can you tell if the two images are two EPI acquisitions with different TE, or the same (ME-EPI) acquisition?

Reply:
I thought the context was that, if all entities but `echo-` match, then they are considered the same acquisition. I tried to give options for indicating two separate acquisitions.

If that's not the case, then I don't know.

Reply:
I agree there's ambiguity in this case, but I see strong parallels with
similar ambiguities we encountered (and decided to live with) in
naturalistic stimuli such as movies.

In that case,  different researchers have interpreted continuous
collections of a movie (for example, the first ten minutes, then the second ten minutes) as corresponding to [_task-movie_run-01_, _task-movie_run-02_] *or*
to [_task-movie01_, _task-movie02_]. In the first case (which I find to be more common), it's unclear how to disambiguate repetitions of the same clip from one another.

You can [find the full discussion here,](https://groups.google.com/forum/m/#!topic/bids-discussion/Lk9Na1) but although I
haven't yet seen this case in a dataset I'd also imagine that `acq` would
be a logical way to distinguish these, or even `run` if they shared the
same task and were collected sequentially.

Reply:
Thank you both, @effigies and @emdupre for the patience for this discussion. I think it is important to understand each others' motivation. Cross-modal discussions are also important to avoid overfitting and redundant solutions.

I can accept ambiguity in the interpretation of the data if there is no way to resolve it, and we should not forget the 80-20 rule (if it is still applicable). We can always attach descriptions and even pipelines to demonstrate the use case of the data.
What I also like to avoid is ambiguity in the spec. If we suggest `acq` or `run` should be used to group data, then we should not introduce grouping suffixes at all, because they can be applied to any modalities.

I do not work much with qMRI data, so I hope someone else from the qMRI community will jump in a correct me if I am wrong. But I also remember discussing suffixes within BEP001, and I was also against it for similar reasons you mentioned here. I admit that acquiring multiple instances of single-echo BOLD with different TEs is not common, and we can probably safely assume that it will not cause an issue. In qMRI, however, there may be more of such cases, as far as I understood. Also, the acquisition of fMRI data is much less diverse, and it can be safely resolved with a few simple entities. In qMRI, however, there is a whole new level of diversity. I myself was also astonished by it. And if we force it to be resolved with entities, then we end up either with complex labels describing many parameters at the same time (e.g. having multiple MT and MPM data at different resolutions), or a multitude of entities, which end up in very long filenames. Worse, we will all know what acquisitions certain combinations of parameters represent, and qMRI people will always use that name to them, which will eventually discourage them from using the spec.

Reply:
Regarding this unfortunate ambiguity, I think that we should require delineation between acquisitions in cases where these multi-contrast (or perhaps better referred to as "within-acquisition") entities are used. This is part two of [my proposed compromise](https://github.com/bids-standard/bids-specification/pull/508#discussion_r449232779). I do not know if it would be backwards incompatible to _require_ that additional entities be used to distinguish acquisitions in this case, but we can at least strongly recommend it until BIDS 2.0.

We could even, perhaps, start referring to entities as either **within-acquisition** or **between-acquisition**, to make the difference clearer. 

We also need to better define "data acquisition", "run", and the idea of "associated scans not acquired in the same acquisition" (e.g., MPM). The idea that associated scans not acquired in the same acquisition could share a suffix and only vary along within-acquisition entities would be a new concept to add to the specification, and would be particularly helpful for qMRI.

> Also, the acquisition of fMRI data is much less diverse, and it can be safely resolved with a few simple entities. In qMRI, however, there is a whole new level of diversity. I myself was also astonished by it. And if we force it to be resolved with entities, then we end up either with complex labels describing many parameters at the same time (e.g. having multiple MT and MPM data at different resolutions), or a multitude of entities, which end up in very long filenames.

I can understand the concept, but am still a little confused by this. This BEP _is_ introducing several new entities, so datatypes within a single multi-contrast protocol are still being distinguished by entities, even if they share the same suffix. I haven't seen cases where grouping suffix results in multiple files that don't use entities to produce unique filenames. While it is true that a truly generic suffix (e.g., `GRE`) and the proposed set of suffixes could not distinguish between certain protocols, that only requires some new suffixes (but not, necessarily, _grouping_ suffixes). I agree that the specification needs those new suffixes, but only in cases where current tools within the specification cannot properly distinguish between the scan types.

> Worse, we will all know what acquisitions certain combinations of parameters represent, and qMRI people will always use that name to them, which will eventually discourage them from using the spec.

I think it's a good idea to use the qMRI/ISMRM folks' terminology when we add new suffixes, but I don't think we should discard the mechanisms already built into the specification when those are sufficient to describe data.

> 3. I am not sure what is the difference between VFA and MEGRE from your POV. Moreover, the issue identified in 2. also applies for MEGRE and MESE.

My interpretation of what @agahkarakuzu said in this thread and on Mattermost was that in MEGRE you basically had a GRE where you could choose the echo times, and that would work fine, while for VFA it was a set protocol/framework. If that's not the case, and VFA is just a GRE with multiple flip angles, then I'd stick with that.

Reply:
I think this thread can be resolved given the discussions we've had in other avenues. The general consensus was that these suffixes would remain, but the idea that multi-file groups would be set based on the _suffix_ (i.e., the idea of a grouping suffix) would be dropped.